src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php

```diff
@@ -25,11 +27,19 @@ class LoggerDataCollector extends DataCollector implements LateDataCollectorInte
 {
     private $logger;
 
-    public function __construct($logger = null)
+    private $filesystem;
+
+    private $cacheDir;
+
+    private $containerBuildLogs = [];
+
+    public function __construct(Filesystem $filesystem, $logger = null, $cacheDir)
```

Comment:
this cannot be required. It is a BC break.

And adding a new argument at the beginning is even worse.